src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -646,32 +663,37 @@ public DenseMatrix calculateStates(JacobianMatrix<DcVariableType, DcEquationType
+        if (contingency.getHvdcIdsToOpen().isEmpty() && contingency.getGeneratorIdsToLose().isEmpty() && contingency.getLoadIdsToLose().isEmpty()) {
             calculateSensitivityValues(factors, factorStates, contingenciesStates, flowStates, contingencyElements,
                     contingency, valueWriter);
         } else {
-            // if we have a contingency including the loss of a DC line or a generator
-            // if we have a contingency including the loss of a generator
+            // if we have a contingency including the loss of a DC line or a generator or a load
 
             Map<LfBus, BusState> busStates = new HashMap<>();
             Set<LfGenerator> participatingGeneratorsToRemove = new HashSet<>();
-            applyInjectionContingencies(network, lfNetwork, contingency, participatingGeneratorsToRemove, busStates, lfParameters);
+            Set<Load> participatingLoadsToRemove = new HashSet<>();
+            applyInjectionContingencies(network, lfNetwork, contingency, participatingGeneratorsToRemove, participatingLoadsToRemove, busStates, lfParameters);
 
             List<ParticipatingElement> newParticipatingElements = participatingElements;
             DenseMatrix newFactorStates = factorStates;
             boolean participatingElementsChanged = !participatingGeneratorsToRemove.isEmpty()
+                || !participatingLoadsToRemove.isEmpty()
                 || (isDistributedSlackOnGenerators(lfParameters) && !contingency.getGeneratorIdsToLose().isEmpty())
-                || (isDistributedSlackOnLoads(lfParameters) && !contingency.getHvdcIdsToOpen().isEmpty());
+                || (isDistributedSlackOnLoads(lfParameters) && (!contingency.getHvdcIdsToOpen().isEmpty() || !contingency.getLoadIdsToLose().isEmpty()));
             if (participatingElementsChanged) {
                 // deep copy of participatingElements, removing the participating LfGeneratorImpl whose targetP has been set to 0
-                newParticipatingElements = participatingElements.stream()
-                    .filter(participatingElement -> !participatingGeneratorsToRemove.contains(participatingElement.getElement()))
-                    .map(participatingElement -> new ParticipatingElement(participatingElement.getElement(), participatingElement.getFactor()))
-                    .collect(Collectors.toList());
+                if (isDistributedSlackOnGenerators(lfParameters)) {
+                    newParticipatingElements = participatingElements.stream()
+                            .filter(participatingElement -> !participatingGeneratorsToRemove.contains(participatingElement.getElement()))
+                            .map(participatingElement -> new ParticipatingElement(participatingElement.getElement(), participatingElement.getFactor()))
+                            .collect(Collectors.toList());
+                } else { // slack distribution on loads.
```

Comment:
Here, I am not sure about this condition: I have to check if LCCs are included in load slack distribution or not.